pkg/zdb/zdb.go

```diff
@@ -98,7 +100,16 @@ func newRedisPool(address string) (*redis.Pool, error) {
 
 	return &redis.Pool{
 		Dial: func() (redis.Conn, error) {
-			return redis.Dial(u.Scheme, host, opts...)
+			con, err := redis.Dial(u.Scheme, host, opts...)
+			if err != nil {
+				return nil, err
+			}
+			_, err = con.Do("AUTH", password)
```

Comment:
This is correct, but now there is an issue regarding "backward compatibility" since on a life node some zdbs will be running already with no protection. so if u fail on AUTH, you need to try to see if authentication is require or not. if not required then u return with no error, but if require u return error 

basically
```
try auth() {
  if err // failed to authorize {
    // check if auth is required (try to do a ping for example)
   if no error {
      //success
   else {
    // possibly other issue or bad passowrd, return error
   }
  }
}
```